prov/verbs/src/fi_verbs.c

```diff
@@ -67,6 +67,71 @@ int fi_ibv_sockaddr_len(struct sockaddr *addr)
 	}
 }
 
+static int fi_ibv_rdm_cm_init(struct fi_ibv_rdm_cm* cm,
+			      const struct rdma_addrinfo* rai)
+{
+	cm->ec = rdma_create_event_channel();
+
+	if (!cm->ec) {
+		VERBS_INFO(FI_LOG_EP_CTRL,
+			"Failed to create listener event channel: %s\n",
+			strerror(errno));
+		return -FI_EOTHER;
+	}
+
+	int fd = cm->ec->fd;
+	int flags = fcntl(fd, F_GETFL, 0);
+	int ret = fcntl(fd, F_SETFL, flags | O_NONBLOCK);
+	if (ret == -1) {
+		VERBS_INFO_ERRNO(FI_LOG_EP_CTRL, "fcntl", errno);
+		return -FI_EOTHER;
+	}
+
+	if (rdma_create_id(cm->ec, &cm->listener, NULL, RDMA_PS_TCP)) {
+		VERBS_INFO(FI_LOG_EP_CTRL, "Failed to create cm listener: %s\n",
+			     strerror(errno));
+		return -FI_EOTHER;
+	}
+
+	if (fi_ibv_rdm_tagged_find_ipoib_addr((struct sockaddr_in*)rai->ai_src_addr,
```

Comment:
Is this check needed here? rai contains the address returned from rdma_getaddrinfo. I believe that should contain the address that belongs to one of the IPoIB interfaces.